### PR TITLE
[cookie-store] Promote stability in test results

### PR DESCRIPTION
**Pull request by jugglinmike**
Allow service worker activation to proceed in the absence of the API
under test. This prevents the workers in failing implementations from
entering the "redundant" state, which can interfere with the harness's
ability to detect all tests.

---

I've observed inconsistent test results from these tests in Safari Technology
Preview 80. This change stabilizes the results and also makes the tests fail
immediately (instead of timing out).

In general, it seems preferable to defer all fallible logic until after the
service worker's been activated. I'm not familiar with the cookie-store draft,
so I've assumed that these operations have to be done during installation.

For cases like that, it might be nice if the harness could be more resilient.
For instance, it might react to the "redundant" state by discarding all tests
(given that it's suspect whether all tests have been transmitted back to the
client) and reporting the "ERROR" status. However, I imagine doing so would
interfere with the service-workers tests that concern redundant workers.
@mattto / @mkruisselbrink do you have any thoughts about that?

### CODE REVIEW

File: cookie-store/serviceworker_cookieStore_subscriptions.js [hunk 1]
```diff
@@ -9,12 +9,18 @@ self.addEventListener('install', (event) => {
...
-    ]);
+    try {
+      await cookieStore.subscribeToChanges([
+        { name: 'cookie-name1', matchType: 'equals', url: '/scope/path1' }]);
+      await cookieStore.subscribeToChanges([
+        { },  // Test the default values for subscription properties.
+        { name: 'cookie-prefix', matchType: 'starts-with' },
+      ]);
+
+    // If the worker enters the "redundant" state, the UA may terminate it
```
Comment (inexorabletash) [inline cookie-store/serviceworker_cookieStore_subscriptions.js]: nit: should probably be indented 2 more spaces

File: cookie-store/serviceworker_cookieStore_subscriptions_basic.js [hunk 2]
```diff
@@ -6,8 +6,14 @@ importScripts("/resources/testharness.js");
 
 self.addEventListener('install', (event) => {
   event.waitUntil((async () => {
-    await cookieStore.subscribeToChanges([
-      { name: 'cookie-name', matchType: 'equals', url: '/scope/path' }]);
+    try {
+      await cookieStore.subscribeToChanges([
+        { name: 'cookie-name', matchType: 'equals', url: '/scope/path' }]);
+
+    // If the worker enters the "redundant" state, the UA may terminate it
```
Comment (inexorabletash) [inline cookie-store/serviceworker_cookieStore_subscriptions_basic.js]: nit: should probably be indented 2 more spaces

File: cookie-store/serviceworker_cookieStore_subscriptions_empty.js [hunk 3]
```diff
@@ -6,7 +6,13 @@ importScripts("/resources/testharness.js");
 
 self.addEventListener('install', (event) => {
   event.waitUntil((async () => {
-    await cookieStore.subscribeToChanges([]);
+    try {
+      await cookieStore.subscribeToChanges([]);
+
+    // If the worker enters the "redundant" state, the UA may terminate it
```
Comment (inexorabletash) [inline cookie-store/serviceworker_cookieStore_subscriptions_empty.js]: nit: should probably be indented 2 more spaces

File: cookie-store/serviceworker_cookieStore_subscriptions_eventhandler_attribute.js [hunk 4]
```diff
@@ -6,8 +6,14 @@ importScripts("/resources/testharness.js");
 
 self.addEventListener('install', (event) => {
   event.waitUntil((async () => {
-    await cookieStore.subscribeToChanges([
-      { name: 'cookie-name', matchType: 'equals', url: '/scope/path' }]);
+    try {
+      await cookieStore.subscribeToChanges([
+        { name: 'cookie-name', matchType: 'equals', url: '/scope/path' }]);
+
+    // If the worker enters the "redundant" state, the UA may terminate it
```
Comment (inexorabletash) [inline cookie-store/serviceworker_cookieStore_subscriptions_eventhandler_attribute.js]: nit: should probably be indented 2 more spaces

File: cookie-store/serviceworker_cookieStore_subscriptions_mismatch.js [hunk 5]
```diff
@@ -6,8 +6,14 @@ importScripts("/resources/testharness.js");
 
 self.addEventListener('install', (event) => {
   event.waitUntil((async () => {
-    await cookieStore.subscribeToChanges([
-      { name: 'cookie-name', matchType: 'equals', url: '/scope/path' }]);
+    try{
+      await cookieStore.subscribeToChanges([
+        { name: 'cookie-name', matchType: 'equals', url: '/scope/path' }]);
+
+    // If the worker enters the "redundant" state, the UA may terminate it
```
Comment (inexorabletash) [inline cookie-store/serviceworker_cookieStore_subscriptions_mismatch.js]: nit: should probably be indented 2 more spaces

File: cookie-store/serviceworker_cookieStore_subscriptions_mismatch.js [hunk 6]
```diff
@@ -6,8 +6,14 @@ importScripts("/resources/testharness.js");
 
 self.addEventListener('install', (event) => {
   event.waitUntil((async () => {
-    await cookieStore.subscribeToChanges([
-      { name: 'cookie-name', matchType: 'equals', url: '/scope/path' }]);
+    try{
```
Comment (inexorabletash) [inline cookie-store/serviceworker_cookieStore_subscriptions_mismatch.js]: nit: space between `try` and `{`

**Review by inexorabletash — COMMENTED** [review]
Seems fine to me. Some formatting nits.

**Comment by jugglinmike** [comment]
Thanks for the review, @inexorabletash.

[Here's an initial idea for addressing this in the harness.](https://github.com/web-platform-tests/wpt/compare/master...bocoup:testharness-sw-redundant) It triggers a harness error if the worker enters the redundant state. That makes the problem more clear, but it still requires timing out, and it's still susceptible to instability. I think the true solution will require a more involved patch which is not something I'd like to embark on without support from the service workers experts.

**Review by inexorabletash — APPROVED** [review]
LGTM, thanks!

**Comment by inexorabletash** [comment]
> ... not something I'd like to embark on without support from the service workers experts

+1; moar qualified eyeballs needed.

**Comment by jugglinmike** [comment]
(discovered while [researching discrepancies in test results for Safari between WPT's various results collectors](https://github.com/web-platform-tests/results-collection/issues/646))

**Pull request merged**